src/ser/mod.rs

```diff
@@ -2,15 +2,27 @@ use serde::{ser, Deserialize, Serialize};
 use std::{
     error::Error as StdError,
     fmt::{Display, Formatter, Result as FmtResult, Write},
+    io,
 };
 
 mod value;
 
+/// A convenience function for writing data to a writer
+pub fn to_writer<W, T: ?Sized>(writer: W, value: &T) -> Result<()>
+where
+    W: io::Write,
+    T: Serialize,
+{
+    { writer }
```

Comment:
why are "{" and "}" needed here?

Reply:
So basically this creates a temporary variable binding (which are always `mut`) and then we chain the `write_all` (which requires `mut`) into that, it's a way to "dodge" the fact we don't ask for a `mut` writer as a parameter for this function.

The reason the function signature itself doesn't ask for a `mut` is because this is the same function signature other `serde` libraries uses:
[serde_json's to_writer](https://docs.serde.rs/serde_json/fn.to_writer.html)
[serde_yaml's to writer](https://docs.serde.rs/serde_yaml/fn.to_writer.html)
[Bincode's seralize_into](https://docs.rs/bincode/1.1.4/bincode/fn.serialize_into.html)

Reply:
I see. I don't think `mut` in a function parameter is a part of the signature. It's purely an implementation detail that we shouldn't be shy of using, if it's relevant.

Reply:
I don't mind changing it to `mut`, gonna amend that now